eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -254,26 +273,70 @@ object Graph {
     }
   }
 
+  // Computes the compound weight for the given @param edge, the weight is cumulative and must account for the previous edge's weight.
+  private def edgeWeight(edge: GraphEdge, prev: RichWeight, isNeighborTarget: Boolean, currentBlockHeight: Long, wr: WeightRatios): RichWeight = {
+    import RoutingHeuristics._
+
+    // Every edge is weighted by funding block height where older blocks add less weight, the window considered is 2 months.
+    val channelBlockHeight = ShortChannelId.coordinates(edge.desc.shortChannelId).blockHeight
+    val ageFactor = normalize(channelBlockHeight, min = currentBlockHeight - BLOCK_TIME_TWO_MONTHS, max = currentBlockHeight)
+
+    // Every edge is weighted by channel capacity, larger channels add less weight
+    val edgeMaxCapacity = edge.update.htlcMaximumMsat.getOrElse(CAPACITY_CHANNEL_LOW_MSAT)
+    val capFactor = 1 - normalize(edgeMaxCapacity, CAPACITY_CHANNEL_LOW_MSAT, CAPACITY_CHANNEL_HIGH_MSAT)
+
+    // Every edge is weighted by its clvt-delta value, normalized
+    val channelCltvDelta = edge.update.cltvExpiryDelta
+    val cltvFactor = normalize(channelCltvDelta, CLTV_LOW, CLTV_HIGH)
+
+    // NB 'edgeCost' includes the amount to be sent plus the fees that must be paid to traverse this @param edge
+    val edgeCost = if (isNeighborTarget) prev.cost else edgeFeeCost(edge, prev.cost)
+
+    val factor = (cltvFactor * wr.cltvDeltaFactor) + (ageFactor * wr.ageFactor) + (capFactor * wr.capacityFactor) match {
+      case 0 => 0.00001 // if the factor turns out to be 0 we default to a very small number to avoid having a weight of 0 and still take into account the cost
```

Comment:
Why not use 1 instead ? this way we revert to exactly the same algorithm as before (i.e use edge costs)

Reply:
Done in 0acbc5d